scripts/react_native_pods.rb

```diff
@@ -337,32 +367,8 @@ def use_react_native_codegen!(spec, options={})
 # See https://github.com/facebook/react-native/issues/31480#issuecomment-902912841 for more context.
 # Actual fix was authored by https://github.com/mikehardy.
 # New app template will call this for now until the underlying issue is resolved.
+#
+# It's not needed anymore and will be removed later
 def __apply_Xcode_12_5_M1_post_install_workaround(installer)
-  # Apple Silicon builds require a library path tweak for Swift library discovery to resolve Swift-related "symbol not found".
-  # Note: this was fixed via https://github.com/facebook/react-native/commit/eb938863063f5535735af2be4e706f70647e5b90
-  # Keeping this logic here but commented out for future reference.
-  #
-  # installer.aggregate_targets.each do |aggregate_target|
-  #   aggregate_target.user_project.native_targets.each do |target|
-  #     target.build_configurations.each do |config|
-  #       config.build_settings['LIBRARY_SEARCH_PATHS'] = ['$(SDKROOT)/usr/lib/swift', '$(inherited)']
-  #     end
-  #   end
-  #   aggregate_target.user_project.save
-  # end
-
-  # Flipper podspecs are still targeting an older iOS deployment target, and may cause an error like:
-  #   "error: thread-local storage is not supported for the current target"
-  # The most reliable known workaround is to bump iOS deployment target to match react-native (iOS 11 now).
-  installer.pods_project.targets.each do |target|
-    target.build_configurations.each do |config|
-      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '11.0'
-      end
-  end
-
-  # But... doing so caused another issue in Flipper:
-  #   "Time.h:52:17: error: typedef redefinition with different types"
-  # We need to make a patch to RCT-Folly - set `__IPHONE_10_0` to our iOS target + 1.
-  # See https://github.com/facebook/flipper/issues/834 for more details.
-  `sed -i -e  $'s/__IPHONE_10_0/__IPHONE_12_0/' #{installer.sandbox.root}/RCT-Folly/folly/portability/Time.h`
+  puts "__apply_Xcode_12_5_M1_post_install_workaround() is not needed anymore"
```

Comment:
hmm, if we pick this fix to 0.66-stable, then we can delete this whole workaround, because it was added only for 0.66.0 RCs

Reply:
Yeah, that's a thought, and seems clean but would be another rc I guess? But I'm not sure - as a personal opinion - what the real benefit is of another rc cycle vs just doing it for a quick 66.1. There are enough other things that need sussing out for 66 in the ecosystem (e.g., a reanimated release) that putting it out there now has value.

Reply:
One thought is that if the previous approach was not as robust: https://github.com/facebook/react-native/commit/329b026f3a48a4228a89696540ba459276f9a2c2 and if this PR solves it properly, it seems like a reasonable thing to use instead. Plus, if we do that, then we don't have to deal with the deprecation of this workaround macro (since it was added only for 0.66.0). So it's slightly better from maintenance vs going to 0.66.1.

But given folks already tried the workaround and it seemed OK for them, I can also see us skipping this commit until the next release (either 0.67, or if there are other "hotfixes" necessary via 0.66.1). Adding a new RC might delay going to stable by ~4 days or so (just for soak time).

Reply:
The known-ugly + release now, vs the unknown-clean (what if there are problem reports etc? I haven't even tested this personally...) that you can still do later. As a release manager I know which way I normally lean (release the tested thing now) but I do feel the tension with a cleaner approach in reach. I've just seen so many delays reaching before I'm timid about it now.

Reply:
OK, thanks for the input. Given this, let's skip this PR for 0.66.0, and we'll figure out next available release to pick it.